tests/test_fitcube.py

```diff
@@ -529,5 +529,58 @@ def test_groupSettings(self):
         self.assertEqual((0.25, False, True), config2.getGroupDataProportion("sides"))
         self.assertEqual((1.0, None, True), config2.getGroupDataProportion("top"))
 
+    def test_preAlignment(self):
+        """
+        Test prealignment step to ensure models at different translation, scale and rotation all return close
+        to same aligned model.
+        """
+        zinc_model_file = os.path.join(here, "resources", "cube_to_sphere.exf")
+        zinc_data_file = os.path.join(here, "resources", "cube_to_sphere_data_random.exf")
+        fitter = Fitter(zinc_model_file, zinc_data_file)
+        self.assertEqual(1, len(fitter.getFitterSteps()))
+        fitter.setDiagnosticLevel(1)
+
+        # Rotation, scale, translation
+        transformationList = [[[ 0.0, 0.0, 0.0 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ math.pi * 20/180, 0.0, 0.0 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ math.pi * 135/180, 0.0, 0.0 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ math.pi * 250/180, math.pi * -45/180, 0.0 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ math.pi * 45/180, math.pi * 45/180, math.pi * 45/180 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ 0.0, 0.0, 0.0 ], 0.05, [ 0.0, 0.0, 0.0]],
+                              [[ math.pi * 70/180, math.pi * 10/180, math.pi * -300/180 ], 0.2, [ 0.0, 0.0, 0.0 ]],
+                              [[ 0.0, 0.0, 0.0 ], 1.0, [ 15.0, 15.0, 15.0 ]],
+                              [[ 0.0, 0.0, 0.0 ], 20.0, [ 50.0, 0.0, 10.0 ]],
+                              [[ math.pi * 90/180, math.pi * 200/180, math.pi * 5/180 ], 1.0, [ -10.0, -20.0, 100.0 ]],
+                              [[ math.pi * -45/180, math.pi * 120/180, math.pi * 10/180 ], 500.0, [ 100.0, 100.0, 100.0 ]]]
+
+        for i in range(len(transformationList)):
+            fitter.load()
+            modelCoordinates = fitter.getModelCoordinatesField()
+            rotation = transformationList[i][0]
+            scale = transformationList[i][1]
+            translation = transformationList[i][2]
+            modelCoordinatesTransformed = createFieldsTransformations(modelCoordinates, rotation, scale, translation)[0]
+            fieldassignment = modelCoordinates.createFieldassignment(modelCoordinatesTransformed)
+            fieldassignment.assign()
+
+            align = FitterStepAlign()
+            fitter.addFitterStep(align)
+            self.assertTrue(align.setAlignGroups(True))
+            self.assertTrue(align.isAlignGroups())
```

Comment:
This should be using setAlignMarkers only. Problem is that the groups centres are all on a line for this model, but there are 4 markers which guarantee a unique solution.
Then the issue is how to check the result as the group centres are again insufficient -- in all cases the final nodes should move to the same coordinates. Must check at least 3 nodes, perhaps best all 8. No need to check derivatives, just coordinate values.